### PR TITLE
UI refactor

### PR DESCRIPTION
**Pull request by 4z0t**
_Something's wrong i can feel it..._
* Use **not IsDestroyed(...)** for controls existing check instead null check
* LayoutHelpers preferable instead direct Set
* **self** in methods (**control** if **self** in outer scope)

### CODE REVIEW

File: lua/ui/maputil.lua [hunk 1]
```diff
@@ -57,11 +57,7 @@ end
 
 -- given a scenario, determines if it can be played in skirmish mode
 function IsScenarioPlayable(scenario)
-    if not scenario.Configurations.standard.teams[1].armies then
-        return false
-    end
-
-    return true
+    return not not scenario.Configurations.standard.teams[1].armies
```
Comment (HUSSAR-mtrela) [inline lua/ui/maputil.lua:60]: this should be simplified to:
```return scenario.Configurations.standard.teams[1].armies```

File: lua/ui/notify/customiser.lua [hunk 2]
```diff
@@ -171,16 +155,15 @@ local function SelectLine(dataIndex)
...
-    if not bgSize then bgSize = 20 end
-    if not bgColor then bgColor = 'FF343232' end
-    if not txtColor then txtColor = UIUtil.categoryTextColor end
-    if not txtSize then txtSize = 18 end
-    if not txt then txt = '?' end
+    bgSize = bgSize or 20
+    bgColor = bgColor or 'FF343232'
+    txtColor = txtColor or UIUtil.categoryTextColor
+    txtSize = txtSize or 18
+    txt = txt or '?'
```
Comment (HUSSAR-mtrela) [inline lua/ui/notify/customiser.lua:162]: good job!

File: lua/ui/uiutil.lua [hunk 3]
```diff
@@ -980,15 +979,16 @@ function CreateWorldCover(parent, colorOverride)
         local index = i
         if GetFrame(index) == parent:GetRootFrame() then
             worldCovers[index] = Bitmap(parent)
-            worldCovers[index].ID = index
-            worldCovers[index].OnDestroy = function(self)
+            local worldCover = worldCovers[index]
```
Comment (HUSSAR-mtrela) [inline lua/ui/uiutil.lua:982]: defining `worldCover` locally on line 982 and again `worldCover` on line 1001 might lead to bugs if someone refactors these code or try to understand it. I suggest to renaming one of these local variables:
```local worldCover = worldCovers[index]```
to something like this:
```local frameCover = worldCovers[index]```

**Review by HUSSAR-mtrela — CHANGES_REQUESTED** [review]

File: lua/ui/maputil.lua (hunk 1 above)
Comment (4z0t) [reply lua/ui/maputil.lua:60]: It must return bool, so, I cheated 😁

File: lua/ui/uiutil.lua (hunk 3 above)
Comment (Garanas) [reply lua/ui/uiutil.lua:982]: I agree.

**Comment by Garanas** [comment]
I'd like to note that I am in favor of several smaller pull requests instead of one larger pull request.

**Comment by 4z0t** [comment]
I don't think it's gonna be that large, currently just removing strange and useless stuff and changing code for better readability

File: lua/ui/maputil.lua (hunk 1 above)
Comment (ChrisKitching) [reply lua/ui/maputil.lua:60]: Maybe consider a function like `Bool()` to perform the conversion, then? Javascript-style implicit conversion to boolean may cause engineer killing sprees.

File: lua/ui/menus/eula.lua [hunk 4]
```diff
@@ -3,7 +3,7 @@
 --* Author: Ted Snook
 --* Summary: Front end display of the EULA
 --*
---* Copyright � 2005 Gas Powered Games, Inc.  All rights reserved.
+--* Copyright � 2005 Gas Powered Games, Inc.  All rights reserved.
```
Comment (ChrisKitching) [inline lua/ui/menus/eula.lua:6]: You saved this file with non-UTF-8 encoding, breaking this character. Reconfigure your editor appropriately ;)

Comment (4z0t) [reply lua/ui/menus/eula.lua:6]: it is utf-8
image

**Comment by 4z0t** [comment]
I also know why would chat crash if you spamed messages a  lot

**Comment by Rackover** [comment]
> I also know why would chat crash if you spamed messages a lot.
Because Mr [@Rackover](https://github.com/Rackover) decided to create for each message received a control...

it's Ms 😂   and do know that I receive this kind of ping by mail directly, so editing my name out doesn't change much...
but yes, i believe this is & was my only PR that ever dealt with Lua code , and supcom's Lua API is not one of my strong suits   Feel free to refactor and shuffle around in the way you feel is fit! ❤️

**Comment by 4z0t** [comment]
Yeah, I think I will make new pr for this one, because it requires significant changes

File: lua/ui/menus/eula.lua (hunk 4 above)
Comment (ChrisKitching) [reply lua/ui/menus/eula.lua:6]: Huh, so why'd this break? :D

**Comment by 4z0t** [comment]
Gonna separate on smaller PRs to prevent overgrowing

**Comment by Garanas** [comment]
Sounds better 😄 .

**Pull request closed without merging**